extensions/HyperVExtension/src/HyperVExtension/Exceptions/HyperVManagerException.cs

```diff
@@ -5,13 +5,15 @@ namespace HyperVExtension.Exceptions;
 
 public class HyperVManagerException : Exception
 {
-    public HyperVManagerException(string? message)
+    public HyperVManagerException(string? message, int hresult = 0)
         : base(message)
     {
+        HResult = hresult;
     }
 
     public HyperVManagerException(string? message, Exception? innerException)
     : base(message, innerException)
     {
+        HResult = innerException?.HResult ?? 0;

```

Comment:
Sanity check, if there isn't an inner exception hresult, we store success?

Reply:
I'd rather call it "default value" in this context.
Throwing a C# exception without setting HResult leaves it initialized to 0 by default. .NET doesn't require HResult to be set since it doesn't serve as an indicator of success/failure (compared to WIN32 APIs, for example). The presence of an exception means there was a failure and default value of HResult or other members means they were not reported.

In context of our telemetry, we have success/failure status as a separate field, so default value of HResult also means that it was not reported.